docs/tiltfile_concepts.md

```diff
@@ -144,38 +144,95 @@ can be expanded and investigated.
 Tilt generates these bundles of work after executing your `Tiltfile`. Some Tiltfile calls
 (e.g. `local_resource`) correspond to a single resource; for other calls (e.g. `docker_build` + `k8s_yaml`), 
 Tilt must join multiple bits of work into a single resource. For Kubernetes resources, Tilt 
-does this assembly by scanning all loaded YAML for any k8s objects that it considers a workload.
-Each of these workloads becomes a Tilt resource. If Tilt finds any image build directives 
-corresponding to an image in a workload, that directive also gets added to that resource. 
-(The assembly logic is similar for Docker Compose resources. For more information, see the 
-[Docker Compose documentation](docker_compose.html).)
-
+does this assembly by scanning all loaded YAML for any k8s objects that it considers a workload
+(i.e. any objects that create pods). Each of these workloads becomes a Tilt resource. If Tilt
+finds any image build directives corresponding to an image in a workload, or any Kubernetes
+objects obviously affiliated with that workload (currently the only eligible objects are
+Services), they get added to that resource. (The assembly logic is similar for Docker Compose
+resources. For more information, see the [Docker Compose documentation](docker_compose.html).)
+
+### Configuring Kubernetes Resources
 In many cases, Tilt's automatic resource assembly logic will be sufficient for you to run
 your app. However, if you need to configure your Kubernetes resources on top of Tilt's
 automatic assembly, you can do so with a call to [`k8s_resource`](api.html#api.k8s_resource).
-Here we'll discuss the most common configuration arguments: `new_name` and `port_forwards`.
-For discussion of other available arguments, see the [api spec](api.html#api.k8s_resource).
 
-`new_name` allows you to specify a new resource name, in case you do not like the automatically generated one.
+Here we'll discuss the most common configuration options for Kubernetes resources:
+`new_name`, `port_forwards`, and `objects`. For discussion of other available arguments, see
+the [API spec](api.html#api.k8s_resource).
+
+`new_name` allows you to specify a new resource name, in case you do not like the
```

Comment:
Maybe mention here, or below when we talk about the objects parameter, how this is also required in some circumstances?

Reply:
good call will do